endpoints/_includes/functions.php

```diff
@@ -746,6 +748,42 @@ function helplineSearch($latitude, $longitude)
     return json_decode(get($search_url));
 }
 
+function BMLTServerIsLocalConfig($latitude, $longitude)
+{
+    $helpline_search_radius = setting('helpline_search_radius');
+    $bmlt_search_endpoint = setting('tomato_url') . "/client_interface/json/?switcher=GetSearchResults&data_field_key=root_server_uri&sort_results_by_distance=1&long_val={LONGITUDE}&lat_val={LATITUDE}&geo_width=" . $helpline_search_radius;
+    $search_url = str_replace("{LONGITUDE}", $longitude, str_replace("{LATITUDE}", $latitude, $bmlt_search_endpoint));
+    $search_results = json_decode(get($search_url));
+    $pinged_BMLT_root = $search_results[0] -> root_server_uri;
+    $bmlt_root = $GLOBALS['bmlt_root_server']."/";
+    if ($pinged_BMLT_root == $bmlt_root) {
+        return true;
+    } else {
+    return false;
+    }
+}
+
+function getHelplineBMLTRootServer($latitude, $longitude, $search_type)
+{
+    if ($search_type == SearchType::VOLUNTEERS) {
+        if (setting('tomato_helpline_fallback') & !BMLTServerIsLocalConfig($latitude, $longitude)) {
+            return setting('tomato_url');
+        }
+    } else if (($search_type == SearchType::MEETINGS) && setting('tomato_meeting_search')) {
```

Comment:
This pull request is almost good to go.  Why are we coupling meeting searches with a function called `getHelplineBMLTRootServer`? 

Secondly, Isn't using the tomato url deliberate enough to say use tomato rather than add a setting?

Reply:
Just to unify the getting of the BMLT root server

And finding that some piece in the most recent commit broke the admin panel and local yap volunteer routing - all the settings on/off work and tested to know what rootserver to hit - but the getservicebodies is not functioning correctly. Some of my assumptions on null arguments seem to be incorrect.

Reply:
As to the tomato setting - just separated out whether someone wants tomato fallback on their helpline versus meeting searches - which could be separated.

Reply:
If that's the case it should be called `getBMLTRootServer()`

There should be no change that broke the admin panel.

Reply:
The setting of `$helpline_bmlt_root_server` and `$bmlt_root_server` is the distinction between meeting searches and helpline lookups.  I"m not understanding why that is not sufficient.

Reply:
I do not see `tomato_meeting_search` as a change that is necessary.  You can choose to use whatever root server you want by setting it.  If you set `helpline_bmlt_root_server` it becomes mutually exclusive.